src/Database/Expression/CommonTableExpression.php

```diff
@@ -30,258 +29,164 @@ class CommonTableExpression implements ExpressionInterface
     /**
      * The CTE name.
      *
-     * @var string|null
+     * @var string
```

Comment:
For this to be true you need to init it as empty string ''

Reply:
I'm not sure what the correct "trick" is here. If you never allow it to be set to null, phpstan and psalm are ok with saying it's required to be set.

I don't mind initializing it to `''` if that's more proper.

Reply:
I can just say it's nullable and add the checks. Any external use of it is hidden now so what's the difference right.

Reply:
Yeah, this is more for internal correctness.
I would probably prefer nullable and string cast where necessary
Or a protected getter that throws exception if needed as string (orFail pendant).

Reply:
Agreed. I see now harm in generating empty values. Users will get a sql error instead of a crash which makes more sense.

Took me a while to simplify it down.